### DISCUSSION TITLE
Apis not working after deployment

### DISCUSSION THREAD

**Discussion by Pratish10** [vercel/next.js#65149] [Help]
### Summary

After deployment some of the apis are not giving any response

### Additional information

I am using nextjs 14 app router 
Logic is under /app/api/getCategories/route.ts

### Example

below is the screenshot for reference
![Screenshot 2024-04-29 104019](https://github.com/vercel/next.js/assets/82207405/a71e45c1-6c4f-4a50-8dcb-e2bb3caecf49)
Above one i am checking after the deployment
![Screenshot 2024-04-29 104042](https://github.com/vercel/next.js/assets/82207405/58157612-f780-4f8d-91f4-16a3a2b44125)
testing on localhost
![Screenshot 2024-04-29 104051](https://github.com/vercel/next.js/assets/82207405/09fd3240-2154-458d-be65-b39d008e565c)
above is the simple db query i am using postgresql and prisma

**Comment by icyJoseph** [comment]
Hi,

Your Route Handler got automatically opted-in to caching. You have to do one of the following to break out, https://nextjs.org/docs/app/building-your-application/routing/route-handlers#opting-out-of-caching:

- Using the Request object with the GET method.
- Using any of the other HTTP methods.
- Using [Dynamic Functions](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#dynamic-functions) like cookies and headers.
- The [Segment Config Options](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#segment-config-options) manually specifies dynamic mode.

This is a feature of Next.js for data that's perceived to be static. For example, it creates a `.json` file, with the outputs of running your Route Handler during build time. This is useful when data doesn't change that often, it it does at all. Next.js interprets that you might be getting such data, if you don't ever use the request object coming in. 

For your case, you can try to add this to the file:

```ts
// your imports 

export const dynamic = 'force-dynamic'

// your route handler
```

**Comment by Pratish10** [comment]
Hi @icyJoseph I have other apis which does not have this issue I am using the same route handler everywhere For I am experiencing the issue for this particular one

**Reply by icyJoseph** [reply]
Does the build environment have access to the db? Is it the correct db?

**Reply by Pratish10** [reply]
The issue is fixed 
Thanks for the help

**Reply by icyJoseph** [reply]
What was the issue?

**Reply by Pratish10** [reply]
I have mention the headers as following
`const headers = {
      'Cache-Control': 'no-store'
    }`
    
Caching concept in nextjs is confusing and I learned it is not possible to opt out of router cache 
According to the article below
https://nextjs.org/docs/app/building-your-application/caching
